Final Improvements

Wikitelmex opened the pull request:
- refactor the files in one module folder
- remake the live demo to can see the new implementations working
- remove hardcoded items
- implement task crud function
- implement completed task clear option
- implement function that update the items indexes
- saves to local storage

wikitelmex1 committed:
Final Improvements

- refactor the files in one module folder
- remake the live demo to can see the new implementations working
- remove hardcoded items
- implement task crud function
- implement completed task clear option
- implement function that update the items indexes
- saves to local storage

rloterh commented on src/index.js:
```diff
@@ -1,11 +1,11 @@
 // eslint-disable-next-line
```
- [x] You have **disabled linter check** on your code. Please enable linter check on your code to ensure that your code is robust code and clean.

Wikitelmex replied:
it was disabled just one line to the lodash library, but anyway, this module is not used and deleted the line and the module

rloterh commented on src/modules/crudtask.js:
```diff
@@ -19,7 +19,7 @@ export function onNonEditable(elementId = 0) {
   elementArray.sort((a, b) => ((a.index > b.index) ? 1 : -1));
-  for (let index = 0; index < elementArray.length; index++) { // eslint-disable-line
+  for (let index = 0; index < elementArray.length; index += 1) {
     elementArray[index].index = index;
```
- [x] New entries **do not match** the property index. The _property index_ must be set to the value of the _new array length_ (i.e. if a 5th task is added to the list, the index of that task should equal to 5).

**Index mismatch in your application:**
![image](https://user-images.githubusercontent.com/12745474/134726115-4e252c94-7738-4264-82b7-ebccc7febfa1.png)

Wikitelmex replied:
updated index in array of tasks

rloterh reviewed (changes requested):
Hi Alex, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Good Points 👍  
- Descriptive Pull Request
- Good ReadMe
- Great UI design
- No linter errors


## Required Changes ♻️
_Kindly check the comments under the review._

<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

wikitelmex1 committed:
changes required

- delete disable linter in line 2
- updated object.index to respond to the project requirements

mikenath223 reviewed (approved):
Hello there  

Your project is complete!  There is nothing else to say other than... it's time to merge it 🛳 
Congratulations! 🎉 

Happy coding. Cheers! 🚀🚀

If you feel something isn't clear enough you can leave a question or comment in the PR thread.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
